autoPyTorch/pipeline/components/setup/lr_scheduler/CosineAnnealingWarmRestarts.py

```diff
@@ -69,13 +72,12 @@ def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[
 
     @staticmethod
     def get_hyperparameter_search_space(dataset_properties: Optional[Dict] = None,
-                                        T_0: Tuple[Tuple[int, int], int] = ((1, 20), 1),
-                                        T_mult: Tuple[Tuple[float, float], float] = ((1.0, 2.0), 1.0)
+                                        n_restarts: Tuple[Tuple, int] = ((1, 6), 3)
```

Comment:
yes, we need to keep it this way as its a part of how we are applying the search space update. 
This tuple is passed by the HyperparameterSearchSpaceUpdates

Reply:
I checked `HyperparameterSearchSpaceUpdates`, but none of the function has returns. 

In any case, it seems this is not package dependency, so we should fix it in the near future, because meta-programming is not intuitive and easy to understand. (e.g. what is T_0[0][0] here even though we use T_0 as an integer in other locations. The name should have some kind of consistency.)

Reply:
Yeah I agree, can you raise an issue for this?

Reply:
Okay, I will do that, thanks:)

Reply:
Hey @ravinkohli, can we change `n_restarts: Tuple[Tuple, int]` to `n_restarts: Tuple[Tuple[int, int], int]`, so we can we consistent with the other modules ?

Reply:
Could we also keep the `T_mul` to 2 and calculate the initial budget `T_0`. If I am not mistaken you can use the formula for the sum of a geometric progression. I think like this it would converge better.